### PR TITLE
Permalinks for taxons

### PR DESCRIPTION
**Pull request by gonzalovilaseca**
This fixes the permalink generation TODO in taxons.
I've basically added some customization to the TreeHander. I couldn't extend that class as it has private attributes, so I had to copy the whole class.

The handler is located in ``src/Sylius/Bundle/TranslationBundle/GedmoHandler``, if there is a better place to put it just let me know.

### CODE REVIEW

File: src/Sylius/Bundle/TranslationBundle/GedmoHandler/TranslationSlugHandler.php [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+     */
+    public static function validate(array $options, ClassMetadata $meta)
+    {
+        if (!$meta->isSingleValuedAssociation($options['relationField'])) {
+            throw new InvalidMappingException("Unable to find tree parent slug relation through field - [{$options['relationParentRelationField']}] in class - {$meta->name}");
+        }
+//     Comprobar que el parent del relation tb es single?
+//        if (!$meta->isSingleValuedAssociation($options['relationParentRelationField'])) {
+//            throw new InvalidMappingException("Unable to find tree parent slug relation through field - [{$options['relationParentRelationField']}] in class - {$meta->name}");
+//        }
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/TranslationBundle/GedmoHandler/TranslationSlugHandler.php]: ?

Comment (gonzalovilaseca) [inline src/Sylius/Bundle/TranslationBundle/GedmoHandler/TranslationSlugHandler.php]: Sorry..I didn't push the latest changes where I adressed this... :-)

**Comment by gonzalovilaseca** [comment]
I'm going to have a look at the failing tests. Will update ASAP.

**Comment by gonzalovilaseca** [comment]
@pjedrzejewski The failing test is this one: 
```
    Scenario: Creating new taxon with existing name under given taxonomy
        Given I am on the page of taxonomy "Category"
          And I follow "Create taxon"
         When I fill in "Name" with "Electronics"
          And I press "Create"
         Then I should be on the page of taxonomy "Category"
          And I should see "Taxon has been successfully created."
          And I should see 9 taxons in the list
```
This test assumes we can have the same taxon name for two different taxons, that would mean two identic permalinks pointing to different taxons, and that is not correct, isn't it?

I don't know how this test was passing even before this PR.

**Comment by michalmarcinkowski** [comment]
@pjedrzejewski @gonzalovilaseca 
I think we should allow for Taxons with the same name, but not under the same parent.

Valid scenario
```
Electronics > PC
Electronics > Featured
Clothes > T-Shirts
Clothes > Featured
```

Invalid scenario
```
Electronics > PC
Electronics > Featured
Clothes > T-Shirts
Clothes > T-Shirts*
Clothes > Featured
```
_*should not be allowed_

If we go this way, the failed scenario is invalid and we should be replaced with similar examples as these above. In that case the permalink will do the validation job and we should only check for its uniqueness.

**Comment by gonzalovilaseca** [comment]
That is how it's set up right now, that's why the test fails. I agree with you, let's see what pawel says.

**Comment by gonzalovilaseca** [comment]
PR updated with tests passing. I added a test that needs to be implemented (I don't have the time right now).

**Comment by pjedrzejewski** [comment]
Thank you Gonzalo! :+1:

**Pull request merged**